ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java

```diff
@@ -35,11 +37,56 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
         mBoardSize = boardSize;
         mSquareLabels = new ArrayList<>(boardSize.width * boardSize.height);
 
+        addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                Rectangle b = e.getComponent().getBounds();
+                if(b.height > b.width){
+                    setMinimumSize(new Dimension(b.width, b.width));
+                    setPreferredSize(new Dimension(b.width, b.width));
+                    setMaximumSize(new Dimension(b.width, b.width));
+                }
+                else{
+                    setMinimumSize(new Dimension(b.height, b.height));
+                    setPreferredSize(new Dimension(b.height, b.height));
+                    setMaximumSize(new Dimension(b.height, b.height));
+                }
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+
+            }
+
+            @Override
+            public void componentShown(ComponentEvent e) {
+
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+
+            }
+        });
+
         GridLayout gridLayout = new GridLayout(mBoardSize.width + 2, mBoardSize.height + 2);
         setLayout(gridLayout);
         createGrid(gridLayout);
     }
 
+    public void updateDimensions(int width, int height){
+        if(height > width){
```

Comment:
Same comment as above. In fact, why not use the same code in both places? No duplication